node/src/chain/acceptor.rs

```diff
@@ -317,11 +316,104 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
-                    if *hash != self.get_curr_hash().await {
+            Payload::Quorum(qmsg) => {
+                let tip_header = self.tip.read().await.inner().header().clone();
+
+                match msg.header.compare_round(tip_header.height + 1) {
+                    // If Quorum is for the current round, we verify it,
+                    // rebroadcast it, and reroute it to Consensus
+                    Status::Present => {
+                        // Verify Attestation
+                        let cur_seed = tip_header.seed;
+                        let cur_provisioners = self
+                            .provisioners_list
+                            .read()
+                            .await
+                            .current()
+                            .clone();
+
+                        let res = verify_att(
+                            &qmsg.att,
+                            qmsg.header.clone(),
+                            cur_seed,
+                            &cur_provisioners,
+                            None,
+                        )
+                        .await;
+
+                        match res {
+                            Ok(_) => {
+                                // Reroute to Consensus
+                                //
+                                // INFO: rebroadcast of current-round Quorums is
+                                // delegated to Consensus. We do this to allow
+                                // iteration-based logic
+                                let task = self.task.read().await;
+                                if task.is_running() {
+                                    debug!(
+                                        event = "Rerouting Quorum to Consensus",
+                                        round = qmsg.header.round,
+                                        iter = qmsg.header.iteration,
+                                    );
+                                    task.main_inbound.try_send(msg);
+                                }
+                            }
+                            Err(err) => {
+                                error!("Attestation verification failed: {err}")
+                            }
+                        }
+                    }
+
+                    // If Quorum is for a past round, we only rebroadcast it if
+                    // it's Valid and for a different candidate than what we
+                    // accepted.
+                    //
+                    // The rationale is that:
+                    //  - Fail Quorums have no influence on past rounds
+                    //  - Valid Quorums for accepted candidates have been
+                    //    already broadcast by us
+                    Status::Past => {
+                        let mut discarded = false;
+                        let mut reason = "";
+
+                        if let Vote::Valid(candidate) = qmsg.vote() {
```

Comment:
Nit: by turning around this check to exclude `Vote::Valid` (and returning early in such a case) we can avoid a level of nesting:

```
if !matches!(qmsg.vote(), Vote::Valid(_)) {
    debug!(
        event = "Quorum discarded",
        reason= "past round",
        round = qmsg.header.round,
        iter = qmsg.header.iteration,
        vote = ?qmsg.vote(),
    );
    return;
}
```